### ISSUE TITLE
False positive: Field has no items() member

### ISSUE THREAD

**Issue by kazaamjt** [PyCQA/pylint#2600]
### Steps to reproduce
```Python
from typing import Dict
from dataclasses import dataclass, field

@dataclass
class TestClass:
	foo: str
	bar: str
	baz_dict: Dict[str, str] = field(default_factory=dict)

	def some_func(self) -> None:
		for key, value in self.baz_dict.items():
			print(key)
			print(value)
```

### Current behavior
Pylint reports:
"Instance of 'Field' has no 'items' member"

### Expected behavior
No error, because the field is a Dict and thus does have the items() member.
(Pylint thinks it's just a Field, but it's the default factory returns a dict)

### pylint --version output
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]

(I also tested with "pip install pylint astroid --pre -U", which resulted in the same thing)

**Comment by PCManticore** [comment]
Thanks for creating an issue @kazaamjt Pylint currently does not fully understand `dataclasses`, we'll probably need inference support in inferring `field` as the type of the value passed via `default_factory`

**Comment by jfthuong** [comment]
Note: of course same behavior with `keys` and `values` :)

**Comment by eduardomezencio** [comment]
While dataclasses are not supported, is there a way to disable certain checks on dataclasses only, using pylintrc?

**Comment by PCManticore** [comment]
You can try `--ignored-classes` to see if it helps.

**Comment by kazaamjt** [comment]
> 
> 
> Note: of course same behavior with `keys` and `values` :)

**Comment by yellowhat** [comment]
Hi,
may I ask to reopen this issue.
I have the following code:

```python
@dataclass
class Case:
    """Case class (group Item)"""

    name: str
    irr: float = 0
    items: List[Item] = field(default_factory=lambda: [])

    def add_item(self, item: Item) -> None:
        """Add an item to the item list."""

        self.items.append(item)
```

With the version 2.10.2, pylint returns:
```console
E1101: Instance of 'Field' has no 'append' member (no-member)
```

Thanks

**Comment by Pierre-Sassoulas** [comment]
Hi @yellowhat, could you open a new issue, please ?

**Comment by josiahcoad** [comment]
@Pierre-Sassoulas @yellowhat still having this issues. Seems like others are too

https://stackoverflow.com/questions/76860888/pydantic-v2-0-0-with-pylint-no-member-warning
